java/src/main/java/science/icebreaker/network/keyword_merge/entities/Keyword.java

```diff
@@ -32,4 +32,8 @@ public void addAlias(String alias) {
         this.aliases.add(alias);
     }
     
+    @Override
+    public boolean equals(Object o) {
+        return (o instanceof Keyword) && this.keyword == ((Keyword)o).keyword;
```

Comment:
Is there a reason why the string comparison is done with `==`? I know that it often works but it's not entirely safe, is it?

Reply:
Oh yes. I'll change it. It should be fine in general since similar string keywords have the same reference.